QuestionAnswering/QA_ICL/README.md

```diff
@@ -1,15 +1,199 @@
+- Includes tools for data generation and preparation
 
 ## Architecture overview
 
-![](uml/system_architecture.png)
+The architecture of the project consists of a frontend (user interface), a backend (handling queries, model inference, and Knowledge Graph access), and additional data generation scripts.
+
+```mermaid
+graph TD
+    subgraph Frontend
+    A[Next.js app]
+    end
+
+    subgraph Backend
+    C[FastAPI app]
+    E[(Redis)]
+    F[Triton inference server]
+    end
+
+    D[(Blazegraph servers)]
+
+    A[Frontend: Next.js app] -->|Sends requests| C
+    C[Backend: FastAPI app] -->|Responds with data| A
+    C -->|Queries| D
+    D[(Blazegraph servers)] -->|Returns data| C
+    C -->|Stores/retrieves data| E
+    C -->|Sends inference requests| F
+    F[Triton inference server] -->|Returns inferences| C
+
+    classDef database fill:#f9f,stroke:#333,stroke-width:2px;
+    class D,E database;
+```
+
+## Key Components
+1. **Frontend**: A Next.js application serving as the user interface.
+2. **Backend**: A FastAPI application handling core logic and data processing.
+    - **Redis**: Used for semantic parsing example retrieval, entity retrieval via fuzzy search, and temporary storage of QA requests for user session management.
+    - **Triton Inference Server**: Manages machine learning model inference.
+3. **Blazegraph Servers**: Store and manage RDF data, accessible via SPARQL protocol.
 
 ## Project Structure
 
-- [`data_generation`](./data_generation): encompases Python scripts to prepare data for entity linking and in-context learning.
-- [`backend`](./backend): Marie's backend
-- [`frontend`](./frontend): Next app for Marie frontend and a minimal mock backend server
+The project is organised into the following main directories:
+
+- `data_generation/`: Python scripts for entity linking and ICL data preparation. The scripts generate three kinds of data:
+  - Lexicon: A collection of ontology representations in texts for classes, predicates, or entities.
+  - Simplified graph schema: Captures node types, edge types, and relation types.
+  - Examples of data request: Pairs of natural language queries and their corresponding data requests for ICL.
+- `backend/`: The backend, called Marie Backend, it contains the FastAPI application serving as the system's backend. It consists of two main components:
+  - `fastapi_app`: A FastAPI application.
+  - `triton_inference_server`: Serves the Sentence-BERT model for generating text embedding.
+- `frontend/`: The frontend, called Marie Frontend, serves as the user interface for interacting with the backend services. Built using Next.js, it communicates with the backend to perform operations such as querying chemical species and filtering zeolites, and displays the results to users in a user-friendly manner.
+
+## Getting Started
+
+### Prerequisities
+ - Docker (recommended for deployment)
+ - Python 3.8 or higher
+ - Node.js and npm (for frontend development)
```

Comment:
Prerequisites should be separate for the deployment and development setups. In fact, we don't install node.js natively on the Digital Ocean machines for deployment.